dace/sourcemap.py

```diff
@@ -237,15 +244,24 @@ def __init__(self, code: str, name: str, target_name: str):
         self.name = name
         self.code = code
         self.map = {'target': target_name}
+        self.codegen_map = {}
 
-    def mapper(self):
-        """ For each line of code retrieve the corresponding nodes
-            and map the nodes to this line
+    def mapper(self, codegen_debug: bool = False):
+        """ For each line of code retrieve the corresponding identifiers
+            and create the mapping
         """
+        cpp_pattern = re.compile(
+            r'(\/\/\/\/__DACE:[0-9]+:[0-9]+:[0-9]+(,[0-9])*)')
+        codegen_pattern = re.compile(
+            r'(\/\/\/\/__CODEGEN;([A-z]:)?(\/|\\)([A-z0-9-_+]+(\/|\\))*([A-z0-9]+\.[A-z0-9]+);[0-9]+)'
```

Comment:
I precompile the RE patterns now so it's only done once. I should have done this before

Reply:
nice :)

Reply:
shouldn't this be part of `self`, though (as a field)? then you wouldn't need to pass it along to any method

Reply:
Sure, could do this, does it make a difference? Just more readable code

Reply:
yes, more readable code and fewer arguments (shorter diff too :) )